app/models/cdo/report/report-type.js

```diff
@@ -44,6 +44,11 @@ function ViewModel (payload, backNav, errors) {
     errors: []
   }
 
+  if ((payload?.dogs?.length ?? 0) === 0) {
+    this.model.reportType.items.splice(0, 1)
+    this.model.reportType.items.splice(1, 1)
+  }
```

Comment:
bit concerned about this - would it not be better to use pure functions and to avoid side affects like this?